src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -14,71 +14,85 @@
 
     private ReadFilterLibrary(){ /*no instance*/ }
 
-    /**
-     * local classes for static read filters
-     */
+    /** Do not filter out any read. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class AllowAllReadsReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){return true;}}
 
+    /** Filter out reads containing skipped region from the reference (CIGAR strings with 'N' operator). */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     //Note: do not call getCigar to avoid creation of new Cigar objects
     public static class CigarContainsNoNOperator extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return ! CigarUtils.containsNOperator(read.getCigarElements());}}
 
+    /** Keep only reads that are first of pair. */
```

Comment:
Well, @magicDGS, I'm learning as I go about these read filters. Towards the end of yesterday's pass, I learned how to get to the SAMRecordToGATKReadAdapter.java doc where, e.g. `isFirstOfPair()` is defined. E.g. 
```
return isPaired() && samRecord.getFirstOfPairFlag();
```

So let's be more exacting and also include the second requirement. This will really help our users _and_ our developers. 

Let's add the (0x1), e.g.:

> Keep only reads that are first of pair (0x1 and 0x40).